### PR TITLE
added Sentiment Analysis Node

### PR DESCRIPTION
**Pull request by AlexTeusz**
Added the Node Sentiment Analysis to the AzureCS Custom Module. With this Node you're able to get the sentiment score of a given sentence in a given language. The result is stored in the Cognigy Context. 

To test this node, upload the AzureCS Custom Module to a project and choose the AzureCS/Sentiment Analysis Node in the Node Picker Menu.

### CODE REVIEW

File: modules/azureCS/src/module.ts [hunk 1]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+
+/**
+ * Returns the sentiment of a given sentence.
+ * @arg {SecretSelect} `secret` The configured secret to use
+ * @arg {CognigyScript} `text` The text to analyse
+ * @arg {Select[af,ar,bn,bs,bg,yue,ca,zh-Hans,zh-Hant,hr,cs,da,nl,en,et,fj,fil,fi,fr,de,el,ht,he,hi,mww,hu,is,id,it,ja,sw,tlh,tlh-Qaak,ko,lv,lt,mg,ms,mt,nb,fa,pl,pt,otq,ro,ru,sm,sr-Cyrl,sr-Latn,sk,sl,es,sv,ty,ta,te,th,to,tr,uk,ur,vi,cy,yau]} `language` the language of the given text
+ * @arg {CognigyScript} `store` Where to store the result
+ * @arg {Boolean} `stopOnError` Whether to stop on error or continue
+ */
+async function sentimentAnalysis(input: IFlowInput, args: { secret: CognigySecret, text: string, language: string, store: string, stopOnError: boolean }): Promise<IFlowInput | {}> {
```
Comment (robinschuster) [inline modules/azureCS/src/module.ts]: use an imperateve name like "analyse sentiments"

File: modules/azureCS/src/module.ts [hunk 2]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+    // Check if secret exists and contains correct parameters
+    if (!args.secret || !args.secret.key || !args.secret.region) return Promise.reject("Secret not defined or invalid.");
+    if (!args.text) return Promise.reject("No text defined.");
+    if (!args.language) return Promise.reject("No language defined.");
+
+    return new Promise((resolve, reject) => {
+        let result = {};
+        let accessKey = args.secret.key;
+
+        const uri = args.secret.region + '.api.cognitive.microsoft.com';
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

File: modules/azureCS/src/module.ts [hunk 3]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+        const path = '/text/analytics/v2.0/sentiment';
+
+        const response_handler = (response) => {
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

File: modules/azureCS/src/module.ts (hunk 1 above)
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

**Comment by AlexTeusz** [comment]
I added all requested changes by @robinschuster

File: modules/azureCS/src/module.ts [hunk 4]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+                result = { "error": err.message };
+                input.context.getFullContext()[args.store] = result;
+                resolve(input);
+            });
+        };
+
+        let get_sentiments = function (documents) {
+            let body = JSON.stringify(documents);
+
+            let request_params = {
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

File: modules/azureCS/src/module.ts [hunk 5]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+            });
+            response.on('error', (err) => {
+                if (args.stopOnError) { reject(err.message); return; }
+                result = { "error": err.message };
+                input.context.getFullContext()[args.store] = result;
+                resolve(input);
+            });
+        };
+
+        let get_sentiments = function (documents) {
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

File: modules/azureCS/src/module.ts [hunk 6]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+            req.end();
+        }
+
+        let documents = {
+            'documents': [
+                { 'id': '1', 'language': args.language, 'text': args.text },
+            ]
+        };
+
+        get_sentiments(documents);
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

File: modules/azureCS/src/module.ts [hunk 7]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+            let request_params = {
+                method: 'POST',
+                hostname: uri,
+                path: path,
+                headers: {
+                    'Ocp-Apim-Subscription-Key': accessKey,
+                }
+            };
+
+            let req = https.request(request_params, response_handler);
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

File: modules/azureCS/src/module.ts [hunk 8]
```diff
@@ -495,4 +495,80 @@ async function textTranslator(input: IFlowInput, args: { secret: CognigySecret,
...
+ * @arg {Boolean} `stopOnError` Whether to stop on error or continue
+ */
+async function sentimentAnalysis(input: IFlowInput, args: { secret: CognigySecret, text: string, language: string, store: string, stopOnError: boolean }): Promise<IFlowInput | {}> {
+    // Check if secret exists and contains correct parameters
+    if (!args.secret || !args.secret.key || !args.secret.region) return Promise.reject("Secret not defined or invalid.");
+    if (!args.text) return Promise.reject("No text defined.");
+    if (!args.language) return Promise.reject("No language defined.");
+
+    return new Promise((resolve, reject) => {
+        let result = {};
```
[the remark this answers is not in the document]
Comment (AlexTeusz) [reply modules/azureCS/src/module.ts]: done

**Pull request merged**